main/src/main/scala/sbt/internal/GlobalPlugin.scala

```diff
@@ -20,6 +20,7 @@ import sbt.internal.util.Attributed
 import Def.{ ScopedKey, Setting }
 import Keys._
 import Configurations.{ Compile, Runtime }
+import sbt.ProjectExtra.{ extract, runUnloadHooks, setProject }
```

Comment:
With the changes to `LoadBuildConfiguration`, it looks like there may need to be an update to the `def build` method, or else global plugins fail to load, due to being passed the wrong `BASE` directory:
```
java.lang.RuntimeException: /Users/julianpeeters/.sbt/2.0.0-alpha6-SNAPSHOT/plugins/global.sbt cannot be mapped using the root paths
Map(
  BASE -> /Users/julianpeeters/Workspace/hello-global,
  SBT_BOOT -> /Users/julianpeeters/.sbt/boot,
  IVY_HOME -> /Users/julianpeeters/.ivy2,
  JAVA_HOME -> /Users/julianpeeters/.sdkman/candidates/java/17.0.1-oracle
)
```
```scala
  def build(base: File, s: State, config: LoadBuildConfiguration): (BuildStructure, State) = {
    val newConverter =
      MappedFileConverter(config.converter.rootPaths2.toMap.updated("BASE", base.toPath), false)
    val newInject =r
      config.injectSettings.copy(global = config.injectSettings.global ++ globalPluginSettings)
    val globalConfig = config.copy(
      injectSettings = newInject,
      pluginManagement = config.pluginManagement.forGlobalPlugin,
      converter = newConverter
    )
```